### ISSUE TITLE
Dependencies specified with Dependency(allow_failure=True) are never enqueued when a job is moved to FailedJobRegistry due to AbandonedJobError

### ISSUE THREAD

**Issue by Marishka17** [rq/rq#2006]
As stated in the documentation, since version 1.11.0 it is possible to enqueue jobs even if their dependency has failed.
It can be specified with *Dependency(allow_failure=True)*. However, that does not work in one case. 
For example, we have 2 jobs `X1` and `X2`. X2 depends on X1 and can be executed if X1 fails. The code is executed in the docker container. If in this case, the docker container is restarted when X1 is running, then X1 will be moved to `FailedJobRegistry` due to AbandonedJobError after the restart. However, there is no code that will start deferred jobs if needed in the `registry.StartedJobRegistry.cleanup`. Looks like this can be fixed by adding `queue.enqueue_dependents(job)` when handling expired jobs.

rq version: **1.15.1**
